torchdynamo/variables/torch.py

```diff
@@ -158,6 +159,22 @@ def call_function(
             and args[0].size is not None
         ):
             return ConstantVariable(product(args[0].size), **options)
+        elif (
+            self.value
+            in (
+                torch._shape_as_tensor,
+                torch.onnx.operators.shape_as_tensor,
```

Comment:
We don't create a look-up table here because we would need to have more general purpose utility function that make sure these ops are getting correct number of arguments with correct type. Since this is one time off kinda thing, its probably ok just special case here?

Reply:
use inspect for that

Reply:
Also how would we incorporate this set of checks in the look-up table?: 
`if (not torch.jit.is_scripting()) and (not torch.onnx.is_in_onnx_export()):`

Are you ok with ugly code where we look up first and then check these options on a special case? Something like:
```
if op in RETURN_TYPE_REWRITE:
    if op in (torch._shape_as_tensor, torch.onnx.operators.shape_as_tensor):
        if (not torch.jit.is_scripting()) and (not torch.onnx.is_in_onnx_export()):
            REPLACE_THIS_OP()
```

Reply:
> use inspect for that

Can we do this in two steps? First step would require that the number of inspect library schema arguments must be the same as that of schema in native-functions.yaml. (aka not allowing variable number of arguments) Second step, we do more general purpose schema matching stuff (similar to what we have in TorchScript) to make sure the op replacement is absolutely safe. @suo What do you think?